src/main/java/io/kestra/plugin/microsoft365/outlook/List.java

```diff
@@ -0,0 +1,263 @@
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "List recent emails from inbox",
+            full = true,
+            code = """
+                id: list_outlook_emails
+                namespace: company.team
+
+                tasks:
+                  - id: list_emails
+                    type: io.kestra.plugin.microsoft365.outlook.List
+                    auth:
+                      tenantId: "{{ secret('AZURE_TENANT_ID') }}"
+                      clientId: "{{ secret('AZURE_CLIENT_ID') }}"
+                      clientSecret: "{{ secret('AZURE_CLIENT_SECRET') }}"
+                      userPrincipalName: "user@example.com"
+                    folderId: "inbox"
+                    top: 10
+                """
+        ),
+        @Example(
+            title = "List emails with filter",
+            full = true,
+            code = """
+                id: list_filtered_emails
+                namespace: company.team
+
+                tasks:
+                  - id: list_emails
+                    type: io.kestra.plugin.microsoft365.outlook.List
+                    auth:
+                      tenantId: "{{ secret('AZURE_TENANT_ID') }}"
+                      clientId: "{{ secret('AZURE_CLIENT_ID') }}"
+                      clientSecret: "{{ secret('AZURE_CLIENT_SECRET') }}"
+                      userPrincipalName: "user@example.com"
+                    folderId: "inbox"
+                    filter: "isRead eq false and receivedDateTime ge {{ now() | dateAdd(-7, 'DAYS') | date('yyyy-MM-dd') }}T00:00:00Z"
+                    top: 50
+                """
+        )
+    }
+)
+public class List extends Task implements RunnableTask<List.Output> {
+
+    @Schema(
+        title = "Authentication configuration",
+        description = "Microsoft Graph authentication settings including tenant ID, client ID, and client secret"
+    )
+    @Valid
+    @NotNull
+    private GraphAuthConfig auth;
+
+    @Schema(
+        title = "Folder ID",
+        description = "Email folder to retrieve messages from (inbox, sentitems, drafts, deleteditems, or folder ID)",
+        defaultValue = "inbox"
+    )
+    private Property<String> folderId;
+
+    @Schema(
+        title = "Filter",
+        description = "OData filter expression to filter the results (e.g., 'isRead eq false', 'from/emailAddress/address eq 'sender@example.com'')"
+    )
+    private Property<String> filter;
+
+    @Schema(
+        title = "Top",
+        description = "Maximum number of messages to return",
+        defaultValue = "50"
+    )
+    private Property<Integer> top;
+
+    @Schema(
+        title = "User email",
+        description = "Email address of the user whose mailbox to access (optional, uses authenticated user if not specified)"
+    )
+    private Property<String> userEmail;
+
```

Comment:
You have to add a `fetchType` property like it's done for instance in [plugin-jdbc](https://github.com/kestra-io/plugin-jdbc/blob/main/plugin-jdbc/src/main/java/io/kestra/plugin/jdbc/AbstractJdbcQuery.java#L82) because you could fetch one, fetch all or simply want to store the huge result amount (what if you get 1 million mails here?) in the internal storage.